internal/context/workspace/domain/workflow/event_handlers.go

```diff
@@ -33,14 +32,14 @@ const (
 )
 
 type WorkflowVersionAddedHandler struct {
-	repo        Repository
-	womtoolPath string
+	repo   Repository
+	reader Reader
 }
 
-func NewWorkflowVersionAddedHandler(repo Repository, womtoolPath string) *WorkflowVersionAddedHandler {
+func NewWorkflowVersionAddedHandler(repo Repository, readerOptions *ReaderOptions) *WorkflowVersionAddedHandler {
 	return &WorkflowVersionAddedHandler{
-		repo:        repo,
-		womtoolPath: womtoolPath,
+		repo:   repo,
+		reader: &WDLReader{readerOptions},
```

Comment:
Just a WDLReader makes the abstration sterile, should be a Reader{} type and can have a NextflowReader{} impl in your case.